R/cleanup.R

```diff
@@ -713,37 +701,37 @@ cleanGeneDescription <- function(prot, column) {
 #' selectLongestDuplicate()
 #' }
 selectLongestDuplicate <- function(prot, column) {
-    col <- sym(column)
-
-    prot$row.orig <- 1:nrow(prot)
-
+    col <- rlang::sym(column)
+    prot <- prot %>% 
+        mutate(row.orig = seq_len(n()))
     # Get list of duplicates
     dups <- prot %>%
-        group_by(AccNum) %>%
-        summarize("count" = n()) %>%
+        group_by(.data$AccNum) %>%
+        summarize(count = n()) %>%
         filter(count > 1) %>%
-        arrange(-count) %>%
-        merge(prot, by = "AccNum")
+        arrange(desc(count)) %>%
+        left_join(prot, by = "AccNum")
 
-    dup_acc <- dups$AccNum
+    dup_acc <- unique(dups$AccNum)
 
-    longest_rows <- c()
-    remove_rows <- c()
+    longest_rows <- integer()
+    remove_rows <- integer()
     for (acc in dup_acc) {
-        dup_rows <- dups %>% filter(AccNum == acc)
+        dup_rows <- dups %>% filter(.data$AccNum == acc)
 
-        longest <- dup_rows[which(nchar(pull(dup_rows, {{ col }})) == max(nchar(pull(dup_rows, {{ col }}))))[1], "row.orig"]
+        longest <- dup_rows$row.orig[which.max(nchar(pull(dup_rows, !!col)))]
```

Comment:
Selecting the Longest Entry:

- >For each duplicate group, entries are compared based on the length of text in the specified column (column). The function identifies the row with the longest text entry using which.max(nchar(pull(...))), an efficient approach for selecting the desired row within each duplicate group.
- >Rows not selected as the longest entries are stored in remove_rows to exclude them later.